File: ProjetoJava/src/polimorfismo.java
class Animal2 {
    public void som() {}
}

class Cachorro2 extends Animal2 {
    public void som() {
        System.out.println("O cachorro faz: Au au!");
    }
}

class Gato2 extends Animal2 {
    public void som() {
        System.out.println("O gato faz: Miau!");
    }
}

class Polimorfismo {
    public static void emitirSom(Animal2[] animais) {
        for (Animal2 animal : animais) {
            animal.som();
        }
    }

    public static void main(String[] args) {
        Animal2[] animais = { new Cachorro2(), new Gato2() };
        emitirSom(animais);
    }
}

File: GoLangProject/polimorfismo.go
package main

import "fmt"

type Animal interface {
	Som()
}

type Cachorro struct{}

func (c Cachorro) Som() {
	fmt.Println("O cachorro faz: Au au!")
}

type Gato struct{}

func (g Gato) Som() {
	fmt.Println("O gato faz: Miau!")
}

func emitirSom(animais []Animal) {
	for _, animal := range animais {
		animal.Som()
	}
}

func main() {
	animais := []Animal{Cachorro{}, Gato{}}
	emitirSom(animais)
}
